universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -60,11 +74,16 @@ protected ReadOnlyApplicantProgramServiceImpl(
     return getBlockAfter(block.getId());
   }
 
-  // TODO(https://github.com/seattle-uat/universal-application-tool/issues/224): Implement checking
-  // blocks for completion
   @Override
   public Optional<Block> getFirstIncompleteBlock() {
-    ImmutableList<Block> currentBlockList = getCurrentBlockList();
-    return currentBlockList.isEmpty() ? Optional.empty() : Optional.of(currentBlockList.get(0));
+    return getCurrentBlockList().isEmpty()
+        ? Optional.empty()
+        : Optional.of(getCurrentBlockList().get(0));
```

Comment:
Suggestion (blocking): as is this would the first block regardless of whether it's incomplete or not I think?

```suggestion
    return getCurrentBlockList().stream().filter(block -> !block.isCompleteWithoutErrors()).findFirst()
```

Reply:
Ah I see what  you're saying - this should not return blocks that were completed in this program. I'll update

Reply:
I would expect that `getCurrentBlockList()` should return blocks completed in this program so applicants can go back?

Reply:
Agreed - `getCurrentBlockList` should contain the blocks an applicant needs to complete for this program and blocks they have already completed. `getFirstIncompleteBlock` should return the first block the applicant needs to complete, right?

Reply:
I think my wording in the first response is confusing - it should say "`getFirstIncompleteBlock` should not return blocks that were completed in this program"

Reply:
Yes! Sorry for the confusion!